src/xcode/ENA/ENA/Source/Scenes/Events/Store/MockEventStore.swift

```diff
@@ -0,0 +1,91 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import OpenCombine
+
+class MockEventStore: EventStoring, EventProviding {
+
+	// MARK: - Protocol EventStoring
+
+	func createTraceLocation(_ traceLocation: TraceLocation) -> VoidResult {
+		traceLocationsPublisher.value.append(traceLocation)
+		return .success(())
+	}
+
+	func deleteTraceLocation(id: String) -> EventStoring.VoidResult {
+		traceLocationsPublisher.value.removeAll { $0.guid == id }
+		return .success(())
+	}
+
+	func deleteAllTraceLocations() -> EventStoring.VoidResult {
+		traceLocationsPublisher.value = ([])
+		return .success(())
+	}
+
+	func createCheckin(_ checkin: Checkin) -> EventStoring.IdResult {
+		checkinsPublisher.value.append(checkin)
+		return .success((checkinsPublisher.value.count - 1))
+	}
+
+	func updateCheckin(id: Int, endDate: Date) -> EventStoring.VoidResult {
+		var checkins = checkinsPublisher.value
+		var checkin = checkins[id]
+		checkin.update(checkinEndDate: endDate)
+		checkins.remove(at: id)
+		checkins.insert(checkin, at: id)
+		checkinsPublisher.send(checkins)
+		return .success(())
+	}
+
+	func updateCheckin(id: Int, targetCheckinEndDate: Date) -> EventStoring.VoidResult {
+		var checkins = checkinsPublisher.value
+		var checkin = checkins[id]
+		checkin.update(targetCheckinEndDate: targetCheckinEndDate)
+		checkins.remove(at: id)
+		checkins.insert(checkin, at: id)
+		checkinsPublisher.send(checkins)
+		return .success(())
+	}
+
+	func deleteCheckin(id: Int) -> EventStoring.VoidResult {
+		checkinsPublisher.value.remove(at: id)
```

Comment:
Shouldn't this search for the `id` like you did in `deleteTraceLocation(id: String)`?

Reply:
Its not a string but an integer. The idea is, that the id is the index of the array.

Reply:
Didn't though about the deletion, which will break the index references. Will change that.